l1-contracts/contracts/bridge/L1SharedBridge.sol

```diff
@@ -0,0 +1,760 @@
+            _l2MessageIndex,
+            _l2TxNumberInBatch,
+            _merkleProof
+        );
+    }
+
+    function claimFailedDepositLegacyErc20Bridge(
+        address _depositSender,
+        address _l1Token,
+        uint256 _amount,
+        bytes32 _l2TxHash,
+        uint256 _l2BatchNumber,
+        uint256 _l2MessageIndex,
+        uint16 _l2TxNumberInBatch,
+        bytes32[] calldata _merkleProof
+    ) public override onlyLegacyBridge {
+        _claimFailedDeposit(
+            true,
+            ERA_CHAIN_ID,
+            _depositSender,
+            _l1Token,
+            _amount,
+            _l2TxHash,
+            _l2BatchNumber,
+            _l2MessageIndex,
+            _l2TxNumberInBatch,
+            _merkleProof
+        );
+    }
+
+    function _claimFailedDeposit(
+        bool _checkedInLegacyBridge,
+        uint256 _chainId,
+        address _depositSender,
+        address _l1Token,
+        uint256 _amount,
+        bytes32 _l2TxHash,
+        uint256 _l2BatchNumber,
+        uint256 _l2MessageIndex,
+        uint16 _l2TxNumberInBatch,
+        bytes32[] calldata _merkleProof
+    ) internal nonReentrant {
+        {
+            bool proofValid = bridgehub.proveL1ToL2TransactionStatus(
+                _chainId,
+                _l2TxHash,
+                _l2BatchNumber,
+                _l2MessageIndex,
+                _l2TxNumberInBatch,
+                _merkleProof,
+                TxStatus.Failure
+            );
+            require(proofValid, "yn");
+        }
+        require(_amount > 0, "y1");
+
+        {
+            bool notCheckedInLegacyBridgeOrNewTransaction;
+            {
+                // Deposits that happened before the upgrade cannot be checked here, they have to be claimed and checked in the legacyBridge
+                bool weCanCheckDepositHere = ((_chainId != ERA_CHAIN_ID) ||
+                    (_l2BatchNumber >= eraIsEthWithdrawalFinalizedStorageSwitchBatchNumber));
+                // Double claims are not possible, as we this check except for legacy bridge withdrawals
+                // Funds claimed before the update will still be recorded in the legacy bridge
+                // Note we double check NEW deposits if they are called from the legacy bridge
+                notCheckedInLegacyBridgeOrNewTransaction = (!_checkedInLegacyBridge) || weCanCheckDepositHere;
+            }
+            if (notCheckedInLegacyBridgeOrNewTransaction) {
+                bytes32 dataHash = depositHappened[_chainId][_l2TxHash];
+                bytes32 txDataHash = keccak256(abi.encode(_depositSender, _l1Token, _amount));
+                require(dataHash == txDataHash, "ShB: d.it not hap");
+                delete depositHappened[_chainId][_l2TxHash];
+            }
+        }
+
+        if (!hyperbridgingEnabled[_chainId]) {
+            // check that the chain has sufficient balance
+            require(chainBalance[_chainId][_l1Token] >= _amount, "ShB n funds");
```

Comment:
If `_l1Token == l1WethAddress` it failed since on weth deposit `chainBalance` is updated for ether not weth. @kelemeno does it count for crit? :)

Reply:
I think we can count it as such, although funds would not be lost, and after we update the failed deposits could be reclaimed.

Reply:
Fair enough

Reply:
Stale as weth removed